### PR TITLE
Add remote tests.

### PR DESCRIPTION
**Pull request by xiaocq2001**

### CODE REVIEW

File: .github/workflows/regression_test.yml [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        output: file
+
+      # Create a new markdown file "code-coverage-summary.md".
+      # Put the coverage summary and report url to the new markdown file.
+      # Write the coverage summary to job summary.
+    - name: Write Code Coverage Report URL
+      run: |
+        cat code-coverage-results.md > code-coverage-summary.md
+        echo -e "\n" >> code-coverage-summary.md
+        echo '[Open Coverage Report](${{ steps.deployment.outputs.page_url }})' >> code-coverage-summary.md
```
Comment (ting-ms) [inline .github/workflows/regression_test.yml:105]: Recomend to remove line 103- line 105 and changd "code-coverage-summray.md" to "code-coverage-results.md".

Comment (xiaocq2001) [reply .github/workflows/regression_test.yml]: Thanks. Updated.

File: test/regression/tx_event_flags.h
Comment (TiejunMS) [inline test/regression/tx_event_flags.h:1]: Do we need ThreadX files in this folder?

Comment (xiaocq2001) [reply test/regression/tx_event_flags.h:1]: I don't think so, I'll remove.

Comment (xiaocq2001) [reply test/regression/tx_event_flags.h:1]: Removed.

File: .github/workflows/regression_test.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        required: false
+        default: 'default_build_coverage'
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  dump_usbx_inputs:
```
Comment (TiejunMS) [inline .github/workflows/regression_test.yml:26]: Do we still need this?

File: .github/workflows/regression_test.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    if: github.event_name == 'workflow_dispatch'
+    permissions:
+      contents: read
+      issues: read
+      checks: write
+      pull-requests: write
+      pages: write
+      id-token: write
+
+    uses: xiaocq2001/threadx/.github/workflows/regression_template.yml@chxiao/test_workflow
```
Comment (TiejunMS) [inline .github/workflows/regression_test.yml:47]: Should template from azure-rtos org be used?

**Review by TiejunMS — APPROVED** [review]

File: .github/workflows/regression_test.yml (hunk 3 above)
Comment (xiaocq2001) [reply .github/workflows/regression_test.yml:47]: Forget to remove, thanks.

File: .github/workflows/regression_test.yml (hunk 2 above)
Comment (xiaocq2001) [reply .github/workflows/regression_test.yml:26]: Will remove

Comment (xiaocq2001) [reply .github/workflows/regression_test.yml]: Removed.

File: .github/workflows/regression_test.yml (hunk 3 above)
Comment (xiaocq2001) [reply .github/workflows/regression_test.yml]: Updated.

File: .gitmodules
Comment (bo-ms) [inline .gitmodules:1]: empty file, can we remove it?

Comment (xiaocq2001) [reply .gitmodules:1]: Yes. It's for submodules but they are removed. I'll remove this file.

Comment (xiaocq2001) [reply .gitmodules:1]: Removed.

**Review by bo-ms — APPROVED** [review]

**Pull request merged**